bool saw_it
float32 x
float32 y
float32 z
bool fire_extinguished